README.md

```diff
@@ -0,0 +1,123 @@
+
+## How to use?
+
+- Import `SlashMenuPlugin` from [`prosemirror-slash-menu`](https://github.com/emergence-engineering/prosemirror-slash-menu-react)
+- Import `ProsemirrorSuggestcatPluginReact` and `promptCommands` from [`prosemirror-suggestcat-plugin-react`](https://github.com/emergence-engineering/prosemirror-suggestcat-plugin-react)
+- Add `SlashMenuPlugin` to your editor with `promptCommands`
+- Create dom reference
+- Add component next to your editor div
+
+```tsx
+import { SlashMenuPlugin } from "prosemirror-slash-menu";
+import {
+  promptCommands,
+  ProsemirrorSuggestcatPluginReact,
+  slashOpeningCondition,
+} from "prosemirror-suggestcat-plugin-react";
+
+const Editor: FC = () => {
+  // Needed for re-renders on every tr.
+  const [editorState, setEditorState] = useState<EditorState>();
+  const [editorView, setEditorView] = useState<EditorView>();
+  const editorRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    const state = EditorState.create({
+      doc: schema.nodeFromJSON(initialDoc),
+      plugins: [
+        SlashMenuPlugin(promptCommands, undefined, slashOpeningCondition, true),
+      ],
+    });
+    const view = new EditorView(document.querySelector("#editor"), {
+      state,
+      dispatchTransaction: (tr) => {
+        try {
+          const newState = view.state.apply(tr);
+          view.updateState(newState);
+          setEditorState(newState);
+        } catch (e) {}
+      },
+    });
+    setEditorView(view);
+    setEditorState(view.state);
+    return () => {
+      view.destroy();
+    };
+  }, [editorRef]);
+
+  const slashMenuPopperRef = useMemo(() => {
+    if (!editorView || !editorView?.state) {
+      return;
+    }
+
+    const currentNode = editorView.domAtPos(
+      editorView.state.selection.to,
+    )?.node;
+
+    if (!currentNode) {
+      return;
+    }
+
+    if (currentNode instanceof Text) {
+      return currentNode.parentElement;
+    }
+
+    return currentNode instanceof HTMLElement ? currentNode : undefined;
+  }, [editorView?.state?.selection, window.scrollY]);
+
+  return (
+    <Root>
+      <StyledEditor id="editor" ref={editorRef} />
+      {editorView && editorView?.state && slashMenuPopperRef && (
+        <ProsemirrorSuggestcatPluginReact
+          editorView={editorView}
+          editorState={editorView?.state}
+          domReference={slashMenuPopperRef}
+        />
+      )}
+    </Root>
+  );
+};
+```
+
+### Props
+
+- `domReference` This is a [Popper `referenceObject` ](https://popper.js.org/docs/v1/#referenceobject) under which the menu and suggestion overlay will appear. In our example it's under the selected paragraph.
+- `editorView` prosemirror EditorView
+- `editorState` prosemirror EditorState TODO Explain more in NOTE
+
+NOTE: The editor state is passed in separately just so that the component re-renders on state change, in the component it is not used, you should always use `view.state` when you use `prosemirror` in your projects.
```

Comment:
I think we don't need to explain this, this component you wrote needs it as a prop, that is all

Reply:
I think you are right, it was an conversation for us but the user doesnt need that.